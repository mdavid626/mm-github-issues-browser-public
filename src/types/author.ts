export type Author = {
  login: string;
};
