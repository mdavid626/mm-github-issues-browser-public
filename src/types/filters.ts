export const stateFilters = ['open', 'closed'] as const;

export type StateFilter = (typeof stateFilters)[number];

export type Filters = {
  search: string;
  state: StateFilter | null;
};
