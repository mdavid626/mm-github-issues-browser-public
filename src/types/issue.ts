import { PageInfo } from './pagination';

export type IssueState = 'OPEN' | 'CLOSED';

export type IssuesQueryItem = {
  id: string;
  createdAt: string;
  title: string;
  state: IssueState;
  number: number;
  url: string;
  author: {
    login: string;
  };
};

export type IssuesQueryResult = {
  repository: {
    issues: {
      nodes: IssuesQueryItem[];
      pageInfo: PageInfo;
      totalCount: number;
    };
  };
};

export type IssueComment = {
  id: string;
  bodyText: string;
  author: {
    login: string;
  };
  createdAt: string;
};

export type IssueQueryItem = {
  id: string;
  createdAt: string;
  title: string;
  state: IssueState;
  number: number;
  url: string;
  bodyText: string;
  author: {
    login: string;
  };
  comments: {
    nodes: IssueComment[];
    pageInfo: PageInfo;
    totalCount: number;
  };
};

export type IssueQueryResult = {
  repository: {
    issue: IssueQueryItem;
  };
};
