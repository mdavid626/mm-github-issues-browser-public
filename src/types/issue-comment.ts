import { Author } from './author';
import { PageInfo } from './pagination';

export type IssueComment = {
  id: string;
  createdAt: string;
  body: string;
  author: Author;
};

export type IssueQueryComments = {
  nodes: IssueComment[];
  pageInfo: PageInfo;
  totalCount: number;
};
