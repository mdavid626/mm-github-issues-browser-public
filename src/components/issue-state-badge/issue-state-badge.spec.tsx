import { cleanup, render, screen } from '@testing-library/react';
import React from 'react';
import { IssueState } from '../../types/issue';
import IssueStateBadge from './issue-state-badge';

describe('issue-state-badge', () => {
  afterEach(cleanup);

  it('should render', () => {
    const { asFragment } = render(<IssueStateBadge state="OPEN" />);
    expect(asFragment()).toMatchSnapshot();
  });

  const stateCases: [IssueState, string][] = [
    ['OPEN', 'open'],
    ['CLOSED', 'closed'],
  ];
  it.each(stateCases)('should render label for %s', (state, label) => {
    render(<IssueStateBadge state={state} />);
    expect(screen.getByText(label)).toBeVisible();
  });
});
