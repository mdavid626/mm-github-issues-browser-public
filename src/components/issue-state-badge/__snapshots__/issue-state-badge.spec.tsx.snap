// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`issue-state-badge should render 1`] = `
<DocumentFragment>
  <div
    class="IssueStateBadge IssueStateBadge--open"
  >
    open
  </div>
</DocumentFragment>
`;
